include/nlohmann/detail/output/serializer.hpp

```diff
@@ -832,7 +832,12 @@ class serializer
 
         // the actual conversion
         // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg,hicpp-vararg)
-        std::ptrdiff_t len = (std::snprintf)(number_buffer.data(), number_buffer.size(), "%.*g", d, x);
+        std::ptrdiff_t len;
+        if (std::is_same<number_float_t, long double>::value) {
+            len = (std::snprintf)(number_buffer.data(), number_buffer.size(), "%.*Lg", d, x);
```

Comment:
I wonder if it would be better like this, to eliminate the duplication of code which could get out of sync:

    std::ptrdiff_t len = (std::snprintf)(number_buffer.data(), number_buffer.size(), 
        std::is_same<number_float_t, long double>::value ? "%.*Lg" : "%.*g", d, x);

Reply:
Yeah, your solution is better